### ISSUE TITLE
Implement row sampling with a SAMPLE statement

### ISSUE THREAD

**Issue by willium** [cwida/duckdb#1188]
When working with large OLAP data, it can often be necessary to sample rows for performance reasons. Thus, I propose that adding a `SAMPLE` clause [like the one in Clickhouse](https://clickhouse.tech/docs/en/sql-reference/statements/select/sample/) would be tremendously useful (and innovative!) for DuckDB. Bonus points for allowing for different [sampling methods](https://en.wikipedia.org/wiki/Sampling_(statistics)#Sampling_methods). 

As an added point, when sampling and calculating [statistical functions](https://github.com/cwida/duckdb/issues/1125#issuecomment-729086398) like standard deviation or variance, it would be possible to infer that one must use the `_samp` variant (and that that `_pop` variant is invalid)

**Comment by Mytherin** [comment]
I think this is a great idea and something we have discussed before. `SAMPLE` is not so different from `LIMIT`, with the exception that `LIMIT` happens before the end of the query (i.e. after any aggregates). Sample needs to be pushed right after the `FROM` clause.

That means that for a query such as this:

```sql
EXPLAIN SELECT a, SUM(b) FROM test WHERE a<12 GROUP BY a SAMPLE 10;
```

The resulting plan will look like this:

```
┌───────────────────────────┐
│       HASH_GROUP_BY       │
│   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
│             #0            │
│          sum(#1)          │
└─────────────┬─────────────┘                             
┌─────────────┴─────────────┐
│         PROJECTION        │
│   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
│             a             │
│             b             │
└─────────────┬─────────────┘                                            
┌─────────────┴─────────────┐
│           FILTER          │
│   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
│            a<12           │
└─────────────┬─────────────┘            
┌─────────────┴─────────────┐
│           SAMPLE          │
│   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
│             10            │
└─────────────┬─────────────┘                                   
┌─────────────┴─────────────┐
│          SEQ_SCAN         │
│   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
│            test           │
│   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
│             a             │
│             b             │
└───────────────────────────┘  
```

The sample operator can then use reservoir sampling to decide which entries to keep or to discard. An important performance optimization here will be lazy vectors, as the lazy vectors will prevent the full scan result from needing to be computed.

**Comment by Mytherin** [comment]
This should be implemented now in #1195. Documentation is [here](https://duckdb.org/docs/sql/samples).
